lib/jobs/ucs-service-profile-discovery.js

```diff
@@ -0,0 +1,197 @@
+// Copyright 2017, EMC, Inc.
+
+'use strict';
+
+var di = require('di'),
+    urlParse = require('url-parse');
+
+module.exports = UcsServiceProfileDiscoveryJobFactory;
+di.annotate(UcsServiceProfileDiscoveryJobFactory, new di.Provide('Job.Ucs.Service.Profile.Discovery'));
+di.annotate(UcsServiceProfileDiscoveryJobFactory, new di.Inject(
+    'Job.Base',
+    'Logger',
+    'Promise',
+    'Assert',
+    'Util',
+    'Services.Waterline',
+    'Services.Encryption',
+    '_',
+    'JobUtils.UcsTool',
+    'Constants'
+));
+
+function UcsServiceProfileDiscoveryJobFactory(
+    BaseJob,
+    Logger,
+    Promise,
+    assert,
+    util,
+    waterline,
+    encryption,
+    _,
+    UcsTool,
+    Constants
+) {
+    var logger = Logger.initialize(UcsServiceProfileDiscoveryJobFactory);
+
+    /**
+     * @param {Object} options task options object
+     * @param {Object} context graph context object
+     * @param {String} taskId running task identifier
+     * @constructor
+     */
+    function UcsServiceProfileDiscoveryJob(options, context, taskId) {
+        UcsServiceProfileDiscoveryJob.super_.call(this,
+                                   logger,
+                                   options,
+                                   context,
+                                   taskId);
+
+        assert.object(this.options);
+        assert.string(this.options.uri);
+        assert.string(this.options.username);
+        assert.string(this.options.password);
+        assert.string(this.options.ucs);
+        var parse = urlParse(this.options.uri);
+        var protocol = parse.protocol.replace(':','').trim();
+        this.settings = {
+            uri: parse.href,
+            host: parse.host.split(':')[0],
+            root: parse.pathname,
+            port: parse.port,
+            protocol: protocol,
+            username: this.options.username,
+            password: this.options.password,
+            ucs: this.options.ucs,
+            verifySSL: this.options.verifySSL || false
+        };
+        this.ucs = new UcsTool();
```

Comment:
With the same variable name `ucs`, but when it is in `this.settings` it means an IP address, however it is a UCS tool instance if it is in `this`